### DISCUSSION TITLE
Tabela não abre todas as linhas

### DISCUSSION THREAD

**Discussion by jornalistaCarlosOsorio** [sqlitebrowser/sqlitebrowser#3091] [Q&A] [answered]
Estou abrindo uma tabela CSV que foi feita no Excel e salva em CSV. A tabela tem 2921 linhas no Excel. Quando vou abrir do DB Browser Sqlite o arquivo aparece com 2429 linhas. 
Por que isso está acontecendo? O que posso fazer para conseguir realizar as operações necessárias?

**Comment by lucydodo** [comment]
Hi. Do the data that have not been loaded have anything in common?

**Reply by jornalistaCarlosOsorio** [reply]
É a mesma tabela.

**Reply by jornalistaCarlosOsorio** [reply]
Só que aparece de uma forma numa e na outra com menos linhas.

**Reply by chrisjlocke** [reply]
(It just shows up one way in one way and the other with fewer lines.)

**Reply by lucydodo** [reply]
Can you send me the CSV file? (seongtaejg at gmail.com)

**Reply by jornalistaCarlosOsorio** [reply]
Enviei a tabela.

**Reply by lucydodo** [reply]
Well.. As a result of running it on the latest nightly build, it loads the data of 2921 normally.
Would you like to try the latest build? https://nightlies.sqlitebrowser.org/latest/

> Tested on macOS Catalina (Version 10.15.7 19H1922)

**Reply by lucydodo** [reply]
I'm glad you got it resolved.

**Reply by jornalistaCarlosOsorio** [reply]
Que bom, que conseguiu, Gostaria de usar a versão mais recente, sim. Vou baixá-la. Obrigado,

**Reply by jornalistaCarlosOsorio** [reply]
Deu erro novamente. Vou colocar o passo a passo em imagem para saber onde estou errando. Obrigado pela ajuda!

**Comment by chrisjlocke** [comment]
(I'm opening a CSV table that was made in Excel and saved in CSV. The table has 2921 rows in Excel. When I open the DB Browser Sqlite the file appears with 2429 lines.
Why is this happening? What can I do to be able to perform the necessary operations?)

Is it possible to identify what record(s) didn't get imported?  If you can find the row, this would help identify the cause, eg, a field didn't have a value, or was out of range, or more than likely, contained more columns than was expected.  This can happen if the delimiters was wrong.

**Reply by chrisjlocke** [reply]
Is the CSV private?  If you can email me db4s-issue3091 [at] chrisjlocke.co.uk, then I can try and import it using the latest nightly version of DB4S.
What version are you using?
What operating system?

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177823655-98322d34-0b7c-4ba0-ae29-04c0f4b82821.png)

**Comment by jornalistaCarlosOsorio** [comment, marked as answer]
Eu importei os dados copiando as colunas de acordo com a tabela no Excel. Então, quando vou salvar, somem muitas colunas. O que pode ser?

**Reply by jornalistaCarlosOsorio** [reply]
Estou usando esta versão: DB Browser para SQLite versão 3.12.2

**Reply by jornalistaCarlosOsorio** [reply]
Uso o sistema Windows.

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177828625-8979d819-b2bc-4660-b687-3f8c4671a7c0.png)

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177828705-f263a2e0-f229-4db3-a2ab-f49f6100b6ca.png)

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177828750-5d98f255-e8ad-4b57-a05e-5d8a4ff32076.png)

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177828805-84c39e06-2454-4601-afac-6850433dde86.png)

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177828847-24e56846-d20f-49a3-9872-d5cd6a7f8842.png)

**Reply by jornalistaCarlosOsorio** [reply]
Repara que a quantidade de linhas é de 2429 e a planilha principal é de 2920. Sinceramente, não sei o que está acontecendo.

**Reply by jornalistaCarlosOsorio** [reply]
Eu estou usando a versão que você sugeriu. 

DB Browser for SQLite Version 3.12.99 (Jun 29 2022)

Compilado para x86_64-little_endian-llp64, rodando em x86_64
Qt Version 5.12.8
Versão do SQLite 3.39.0.

**Comment by chrisjlocke** [comment]
> I imported the data by copying the columns as per the table in Excel.

![image](https://user-images.githubusercontent.com/7872218/177824652-4364af75-8555-494a-9d73-7f3fe6daccf7.png)

You can click 'File', 'Import' and 'Table from CSV file'
From Excel, you can 'Save As' a CSV file.

**Reply by jornalistaCarlosOsorio** [reply]
Faço isso, mas somem diversas linhas. 
Uma pergunta. Na primeira coluna, vem assim: Escola: 0101001 - EM VICENTE LICINIO CARDOSO Esses dois pontos podem estar interferindo na importação?

**Comment by chrisjlocke** [comment]
I get 2920 rows.

![image](https://user-images.githubusercontent.com/7872218/177831967-72737d8b-2b8e-4c2f-957d-5293a8d87d5d.png)

When you import, what settings do you use?

![image](https://user-images.githubusercontent.com/7872218/177832055-f44cff36-c118-49a6-adf0-cbbe51d5e618.png)

Field seperator as **,** and quote character as **"**

**Reply by chrisjlocke** [reply]
I'm on the wrong PC so can't do a screencast, but don't think it'd show anything worthwhile other than the above import settings.

**Reply by jornalistaCarlosOsorio** [reply]
![image](https://user-images.githubusercontent.com/76069092/177842425-2ce92e4b-0a8f-4427-97bc-b1175c5afe61.png)

**Reply by jornalistaCarlosOsorio** [reply]
Infelizmente, continua dando erro.

**Comment by chrisjlocke** [comment]
I'm guessing its a regional setting.  What import settings are you using?
Is your field seperator a comma and is your quote character a quotation?
In the UK, a thousand is separated by a comma, and a fraction is delimited by a period, so 1000.1 could be 1,000.1
I'm guessing for Portuguese, uyou have a period as a separator, so 1000.1 would be 1.000,1
This obviously breaks the CSV import.  
I couldn't see any values like that in the CSV, but ... well...

You kindly emailed me the CSV file.  Could you email me the database you've created with 2429 rows - I'd like to see what rows are missing - this might help explain whats going on.
Thanks!

**Reply by jornalistaCarlosOsorio** [reply]
Olá, tudo bem? O arquivo em questão foi aberto normalmente no computador de minha casa. Não estou conseguindo abrir esse arquivo em meu notebook. Encaminhei o arquivo para seu email.

Muito obrigado pelo apoio,

**Reply by chrisjlocke** [reply]
> I can't open this file on my notebook. 

Do you know why?  Because of an error?

**Comment by jornalistaCarlosOsorio** [comment]
Prezados, resolvi o problema. Apaguei todos os registros da máquina pelo REGEDIT. Então, baixei a versão mais atualizada e, desta vez, estou conseguindo abrir a planilha completa. 
Desta forma, gostaria de agradecer o apoio para resolver esse problema.

Um grande abraço,

**Reply by chrisjlocke** [reply]
I'm unsure what you deleted via Regedit (some settings?) but glad it solved your problem. 😃

**Reply by jornalistaCarlosOsorio** [reply]
Deu certo. Já fiz isso em outras ocasiões com outros programas. Enfim, muito obrigado mais uma vez. Tudo de bom!  😀
